libnmstate/schemas/operational-state.yaml

```diff
@@ -177,6 +178,32 @@ definitions:
           required:
             - id
             - base-iface
+  interface-vxlan:
+    rw:
+      properties:
+        type:
+          type: string
+          enum:
+            - vxlan
+        vxlan:
+          type: object
+          properties:
+            vni:
+              type: integer
+              minimum: 0
+              maximum: 16777216
+            group:
+              type: string
+            dstport:
+              type: integer
+            base-iface:
+              type: string
+          required:
```

Comment:
Although it is correct, it will be better to raise an error in the code for required fields. Currently, the python json-schema is not that nice when it fails (the output is hard to interpret by a common user).

Reply:
+1

Reply:
This would be done in the implementation PR, that's right ?

Reply:
This is just an opinion, please take it with plenty of salt.

> Although it is correct, it will be better to raise an error in the code for required fields. Currently, the python json-schema is not that nice when it fails (the output is hard to interpret by a common user).

While I agree that the schema's validation errors could be almost considered obfuscation, they still have somewhere a decent indication of the error. Couldn't we wrap those errors, and provide decent info from them instead ? 

Or do we use the schema just to have conversations about the data model ?

Reply:
@maiqueb I forgot it, let me add it.

Reply:
> This would be done in the implementation PR, that's right ?

Yes, we do such things in the `validator` module.

> Couldn't we wrap those errors, and provide decent info from them instead ?

Interpretting the schema error seems more complex that just handling this in the code.

> Or do we use the schema just to have conversations about the data model ?

Schemas can be used for several objectives, one is indeed to describe the data model formally, another is to generate client code (or server code, like for an http service). Documentation generation is another option.
But for the current discussion, only very basic stuff we enforce through the schema.